### ISSUE TITLE
Use unl0kr for easy mobile disk encryption unlocking

### ISSUE THREAD

**Issue by memoryfile** [ublue-os/bazzite#464]
Following up from a discord discussion:

https://gitlab.com/cherrypicker/unl0kr

"looks like it supports drm, that's good 
fedora doesn't ship fbdev
so that was previously holding us back"

![breezy-dark-1920x1080](https://github.com/ublue-os/bazzite/assets/96447025/9331c98c-b908-4d8c-a6d4-5a1ecc4d2787)

**Comment by KyleGospo** [comment]
@EyeCantCU This is building but I'm having some trouble w/ hooking it up in initramfs/dracut. If you (or anyone else) have time, take a look at: https://copr.fedorainfracloud.org/coprs/kylegospo/unl0kr/ in a VM, immutable or otherwise.

SimpleDRM is re-enabled as of Bazzite F39 so once we get this working we can actually ship it on Deck.

**Comment by EyeCantCU** [comment]
Will certainly investigate this

**Comment by RodoMa92** [comment]
This looks amazing, I'll try to take a look at it sooner. No promises tho :P

**Comment by jaduncan** [comment]
This is marked as help wanted - what would be useful?

**Comment by KyleGospo** [comment]
> This is marked as help wanted - what would be useful?

Anything that furthers https://github.com/ublue-os/bazzite/issues/464#issuecomment-1808809849 would be greatly appreciated.

**Comment by memoryfile** [comment]
Just a heads up, unl0kr moved locations https://fosstodon.org/@h3nn3s/112141302068083748

**Comment by bjoern-tantau** [comment]
I got it to work on OpenSUSE. I've made a pull request at https://github.com/KyleGospo/unl0kr/pull/2.

The patch to unl0kr is actually not needed, it works with or without. I also did not test compiling it using the spec file because I frankly don't know how, I did everything by hand. But I guess that it will work.

**Comment by JoshuaFern** [comment]
I was able to use unl0kr with Jovian NixOS just fine but the screen was in the wrong orientation.

In the mean time I've been using deckbd with great success https://github.com/Ninlives/deckbd

**Comment by e-minguez** [comment]
What about this one? https://github.com/pmkap/deckrypt

**Comment by timhae** [comment]
@JoshuaFern would you mind sharing your config? your [nix config](https://github.com/JoshuaFern/nixos-configuration/commits/master/) was last updated two years ago

**Comment by ticky** [comment]
Is there a reason the lack of onscreen keyboard at this stage isn't documented as a caveat? It seems weird that the docs just mention you can choose to enable disk encryption in the handheld guide, but don't mention it's a potential footgun:

> - Optionally encrypt the drive with a password if desired.
>   - If you lose this password, then it cannot be decrypted.

https://universal-blue.discourse.group/t/bazzite-inital-setup-and-installation-for-the-steam-deck/1143

**Comment by atharen** [comment]
**TL;DR: is unl0kr meant to be available in Bazzite? How is it meant to be enabled if so?**

[The `unl0kr` `spec_file` exists](https://github.com/ublue-os/bazzite/tree/main/spec_files/unl0kr) and [the v3.0.0 release notes](https://github.com/ublue-os/bazzite/pull/979#issue-2243873682):
> deck: Add unl0kr for password entry on handhelds & tablets ([d0d9088](https://github.com/ublue-os/bazzite/commit/d0d9088d27c5453ba0826394b43765c4b5fcd235))

Made a clean installation of Bazzite on Deck, and not enabled by default.

Is there a `ujust`  script I'm missing? 
Did try _systemctl enable [unl0kr-ask-password.service](https://github.com/ublue-os/bazzite/blob/main/spec_files/unl0kr/unl0kr-ask-password.service)_, but was not available.

Could not find anything in the documentation, [presumably here](https://github.com/KyleGospo/docs.bazzite.gg/tree/main/src/Advanced).

**Comment by ticky** [comment]
@KyleGospo @EyeCantCU sorry to @ you directly, but would you mind weighing in on this? Is it intended to be usable or is it still experimental?

**Comment by HugKitten** [comment]
Someone apparently got it working on arch, so maybe this can help anyone who is trying to get it working on bazzite (don't count on it):
https://bbs.archlinux.org/viewtopic.php?id=305183
Also keep in mind that they are using mkinitcpio not dracut

**Comment by OisinHick** [comment]
Has there been any progress on this? I just installed bazzite and I am now trying to solve this issue. Might use the TPM chip for decryption but would prefer not to. Considering following this guide https://www.reddit.com/r/Bazzite/comments/1f8iljz/encrypted_data/ using the tpm 2.0 core built in to legion go - just in case anyone else needs a workaround.

**Comment by falhumai96** [comment]
What’s the current status of this issue?

This is a **major requirement** for touch-only handheld devices that want to benefit from the advertised encryption advantages over the original SteamOS. Relying solely on TPM2 for automatic decryption is not sufficient.

Integrating unl0kr as the virtual touch keyboard for unlocking LUKS is essential for usability and security on these devices. Please consider prioritizing this issue.

**Comment by ispaure** [comment]
I would also heavily appreciate this feature for use with my ROG Ally X
